boa_ast/src/function/class.rs

```diff
@@ -505,3 +528,52 @@ impl VisitWith for ClassElement {
         }
     }
 }
+
+/// A private name as defined by the [spec].
+///
+/// [spec]: https://tc39.es/ecma262/#sec-private-names
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "fuzz", derive(arbitrary::Arbitrary))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct PrivateName {
+    /// The `[[Description]]` internal slot of the private name.
+    description: Sym,
+
+    /// The indices of the private name are included to ensure that private names are unique.
+    pub(crate) indices: (usize, usize),
```

Comment:
Seeing the spec about private names it seems that `PrivateName`s should be globally unique, but `indices` are only unique in the context of a single `Context`, right? I was thinking that the global uniqueness property is probably to restrict the access of private names from objects that share the same `indices` but don't belong to the same `Context`, but to me it seems like this doesn't prevent that? I ask this because I don't know if you've already considered that.

Reply:
There is still some uniqueness missing with this implementation. We indeed have to add some runtime unique identifier to separate `PrivateName` of multiple objects that are created from the same class. I think this will probably be some kind of `Context` wide index. As far as different `Context`s go, I think that would need some further work as our `Context` is currently also a `Realm`. If we have figured out how we can separate those, the `Context` should be the outermost api to the engine and imo should the the uniqueness-border.

I left the runtime uniqueness out of the PR to limit the scope somewhat.

Reply:
It's okay! Can you add a `TODO` describing this then? It would be good to track this for when we implement proper Realms.